tests/transform_dialect/cuda/vecadd2d_codegen_spec.mlir

```diff
@@ -0,0 +1,12 @@
+transform.structured.canonicalized_sequence failures(propagate) {
+^bb1(%varop: !pdl.operation):
+  %generic = transform.structured.match ops{["linalg.generic"]} in %varop
+  %foreach, %tiled = transform.iree.tile_to_foreach_thread_and_workgroup_count_region %generic tile_sizes [4, 4] ( mapping = [#gpu.block<x>, #gpu.block<y>] )
```

Comment:
Can we use clearly different sizes here to avoid the corner case of "it works because of magic constants" ?
E.g. `[4, 4]` -> `[2, 3]`

Reply:
It does not work with `[2,3]` and I have no idea why right now. It skips some iterations.  I am not %100 sure, but this can be a different bug. 

For `tile_sizes [4, 4]`
Output is `512x4xf32=[2 2 2 2][2 2 2 2][2 2 2 2]...` which is correct

For  `tile_sizes [2, 3]`
`512x4xf32=[0 0 0 0][0 0 0 0][0 0 0 0]...` is wrong

For `tile_sizes [4, 3]`
`[2 2 2 0][2 2 2 0][2 2 2 0]....` is wrong